Analytics/Classes/Internal/SEGUserDefaultsStorage.h

```diff
@@ -0,0 +1,20 @@
+//
+//  SEGUserDefaultsStorage.h
+//  Analytics
+//
+//  Created by Tony Xiao on 8/24/16.
+//  Copyright © 2016 Segment. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "SEGStorage.h"
+
+@interface SEGUserDefaultsStorage : NSObject <SEGStorage>
+
+@property (nonatomic, strong, nullable) id<SEGCrypto> crypto;
```

Comment:
This makes it difficult to be backward compatible with old storage mechanisms. In particular for user default based storage we don't always storage `NSData` objects, we directly store `NSString` and etc.